### PR TITLE
optimize: modify no available server error more clearly, and fixed NP…

### PR DESCRIPTION
**Pull request by helloworlde**
### Ⅰ. Describe what this PR did
1. Print expect server service name instant of `no available server to connect.` when could found server in registry center
2. Fixed issue of NPE when get server service instances return empty.

### Ⅳ. Describe how to verify it
1. Modify server service group name not `default`.
2. Start a service with using registry center.
3. When service can't find `default` service, will print log like `no available service 'default' found, please make sure registry config correct`

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java [hunk 1]
```diff
@@ -209,15 +212,15 @@ private Channel doConnect(String serverAddress) {
...
-        List<InetSocketAddress> availInetSocketAddressList = RegistryFactory.getInstance().lookup(
-            transactionServiceGroup);
-        if (!CollectionUtils.isEmpty(availInetSocketAddressList)) {
-            for (InetSocketAddress address : availInetSocketAddressList) {
-                availList.add(NetUtil.toStringAddress(address));
-            }
+        List<InetSocketAddress> availInetSocketAddressList = RegistryFactory.getInstance()
+                                                                            .lookup(transactionServiceGroup);
+        if (CollectionUtils.isEmpty(availInetSocketAddressList)) {
+            return null;
```
Comment (xingfudeshi) [inline core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java]: please return a emtpy list to avoid NPE.

Comment (helloworlde) [reply core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java]: Had modified, the caller method has judgement for null or empty too.

File: discovery/seata-discovery-core/src/main/java/io/seata/discovery/registry/RegistryService.java [hunk 2]
```diff
@@ -88,4 +88,12 @@
...
      */
     void close() throws Exception;
+
+    /**
+     * Get current service group name
+     *
+     * @param key service group
+     * @return the service group name
+     */
+    String getServiceGroup(String key);
```
Comment (slievrly) [inline discovery/seata-discovery-core/src/main/java/io/seata/discovery/registry/RegistryService.java]: I think you should define a default method.

Comment (helloworlde) [reply discovery/seata-discovery-core/src/main/java/io/seata/discovery/registry/RegistryService.java]: Had replaced.

**Pull request merged**